app-creator/client/src/widgets/dropzone-widget/dropzone-widget.ts

```diff
@@ -132,15 +131,20 @@ export class Dropzone extends LitElement {
     clone.id += `-${store.getState().widgetIDs[widget.id]}`;
 
     // Return early if the widget has been added to another panel earlier.
-    const template = store.getState().template.widgets;
-    for (const id in template) {
+    const widgets = store.getState().template.widgets;
+    for (const id in widgets) {
       if (
-        typeof template[id] === 'object' &&
-        !Array.isArray(template[id]) &&
+        typeof widgets[id] === 'object' &&
+        !Array.isArray(widgets[id]) &&
         id !== parent!.id &&
-        template[id].children.includes(clone.id)
+        widgets[id].children.includes(clone.id)
       ) {
-        return;
+        const widget = widgets[clone.id].widgetRef;
+        const draggable = widget.parentElement as DraggableWidget;
+        const dropzone = draggable.parentElement as Dropzone;
+        dropzone.removeChild(draggable);
+        const childrenIDs = dropzone.getChildrenIds();
```

Comment:
Treat abbreviations like acronyms in names as whole words, i.e. use childrenIds, not childrenIDs, unless required by a platform name (e.g. XMLHttpRequest).